sstables/mp_row_consumer.hh

```diff
@@ -979,10 +972,11 @@ public:
 
     virtual ~mp_row_consumer_m() {}
 
-    proceed push_ready_fragments() {
+    // See the RowConsumer concept
+    void push_ready_fragments() {
         if (!_mf_filter || _mf_filter->out_of_range()) {
             _reader->on_out_of_clustering_range();
-            return proceed::no;
+            return;
         }
 
         auto maybe_push = [this] (auto&& mfopt) {
```

Comment:
nit: We don't need maybe_push any more. We can just drop the lambda and execute if(mfopt) {...} directly.

Reply:
That was true before the patch as well. I can do this cleanup as a follow-up.